src/hooks/useQuery.ts

```diff
@@ -68,16 +60,18 @@ export const useQuery = <T = any, V = object>(
         })
       );
     },
-    [args.pause, args.requestPolicy, client, request, setState]
+    [args.requestPolicy, client, request, setState]
   );
 
-  // This calls executeQuery immediately during the initial mount and
-  // otherwise behaves like a normal useEffect; We call executeQuery
-  // everytime it, i.e. its input like request, changes
   useImmediateEffect(() => {
+    if (args.pause) {
```

Comment:
Nice I like this. So the flow here is:

- We check `args.pause` on initial mount to handle the case where a user just wants the `executeQuery` fn.
- Also check it on all subsequent changes and prevent queries when things update.

❤️Love it. My one question is should L69, if it's being treated as the cleanup function, be written as `() =>  setState(s => ({ ...s, fetching: false }));`? Seems we're returning `void` currently rather than `() => void`.